components/director/internal/domain/labeldef/service.go

```diff
@@ -128,6 +131,18 @@ func (s *service) Update(ctx context.Context, def model.LabelDefinition) error {
 	return nil
 }
 
+func (s service) Upsert(ctx context.Context, def model.LabelDefinition) error {
+	def.ID = s.uidService.Generate()
```

Comment:
That ID will be used in case there is not an already existing record. We could use the Create function but we would need to make an Exist call prior to calling Create. That way we would need to execute 2 database calls while using the Upsert we only make 1.

Reply:
`s.uidService.Generate()` will always generate and put a non-existing GUID into `def.ID`. In which scenario will this `Upsert` function act as an update rather than a create?